Fix test_ind_worker_queue by setting max_num_worker based on system resource

ejguan opened the pull request:
Fixes #63657

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63779 Fix test_ind_worker_queue by setting max_num_worker based on system resource**

Prevent freezing test for CI.

Confirm this test is flacky due to the limitation of CPU per machine. Did reproduce the hanging workers with ASAN build of PyTorch periodically.

After this PR, the number of worker will be limited based on OS.


Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

cc @SsnL @VitalyFedyunin @ejguan

ejguan committed:
Fix test_ind_worker_queue by setting max_num_worker based on system resource

[ghstack-poisoned]

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"




[ghstack-poisoned]

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"

Fixes #63657




[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"

Fixes #63657


Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"

Fixes #63657


Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"

Fixes #63657


Prevent freezing test for CI.

Confirm this test is flacky due to the limitation of CPU per machine. Did reproduce the hanging workers with ASAN build of PyTorch periodically.

After this PR, the number of worker will be limited based on OS.


Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

[ghstack-poisoned]

VitalyFedyunin commented on test/test_dataloader.py:
```diff
@@ -2288,8 +2288,23 @@ def _run_ind_worker_queue_test(self, batch_size, num_workers):
+            max_num_workers = 1
+
         for batch_size in (8, 16, 32, 64):
-            for num_workers in range(1, 6):
```
nit: min(6, max_num_workers)

ejguan committed:
Update on "Fix test_ind_worker_queue by setting max_num_worker based on system resource"

Fixes #63657


Prevent freezing test for CI.

Confirm this test is flacky due to the limitation of CPU per machine. Did reproduce the hanging workers with ASAN build of PyTorch periodically.

After this PR, the number of worker will be limited based on OS.


Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).
